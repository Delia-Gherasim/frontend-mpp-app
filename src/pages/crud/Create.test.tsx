import {render} from '@testing-library/react';
import {describe, expect, it} from 'vitest';
import AddDevice from './Create';

describe('AddDevice component', () => {
    it('renders correctly', () => {
        const {asFragment} = render(
            <AddDevice
                onBackButtonHandle={() => {}}
                onSaveButtonHandler={() => {}}
            />,
        );
        expect(asFragment()).toMatchSnapshot();
    });
});
