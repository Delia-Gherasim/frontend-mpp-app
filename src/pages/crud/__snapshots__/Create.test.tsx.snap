// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Add > should render the Home component 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="container"
      >
        <h3
          class="title"
        >
          Add New Device
        </h3>
        <form
          class="form-container"
        >
          <div>
            <label>
              Type:
            </label>
            <input
              type="text"
              value=""
            />
          </div>
          <div>
            <label>
              Brand:
            </label>
            <input
              type="text"
              value=""
            />
          </div>
          <div>
            <label>
              Owner:
            </label>
            <input
              type="text"
              value=""
            />
          </div>
          <div>
            <label>
              Accessories:
            </label>
            <input
              type="checkbox"
            />
          </div>
          <div>
            <label>
              Warranty:
            </label>
            <input
              type="checkbox"
            />
          </div>
          <div>
            <label>
              Date:
            </label>
            <input
              data-date=""
              data-date-format="DD MMMM YYYY"
              type="date"
            />
          </div>
          <div
            class="buttons"
          >
            <input
              type="button"
              value="Back"
            />
            <input
              type="button"
              value="Save"
            />
          </div>
        </form>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="container"
    >
      <h3
        class="title"
      >
        Add New Device
      </h3>
      <form
        class="form-container"
      >
        <div>
          <label>
            Type:
          </label>
          <input
            type="text"
            value=""
          />
        </div>
        <div>
          <label>
            Brand:
          </label>
          <input
            type="text"
            value=""
          />
        </div>
        <div>
          <label>
            Owner:
          </label>
          <input
            type="text"
            value=""
          />
        </div>
        <div>
          <label>
            Accessories:
          </label>
          <input
            type="checkbox"
          />
        </div>
        <div>
          <label>
            Warranty:
          </label>
          <input
            type="checkbox"
          />
        </div>
        <div>
          <label>
            Date:
          </label>
          <input
            data-date=""
            data-date-format="DD MMMM YYYY"
            type="date"
          />
        </div>
        <div
          class="buttons"
        >
          <input
            type="button"
            value="Back"
          />
          <input
            type="button"
            value="Save"
          />
        </div>
      </form>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`AddDevice component > renders Adding correctly 1`] = `
<DocumentFragment>
  <div
    class="container"
  >
    <h3
      class="title"
    >
      Add New Device
    </h3>
    <form
      class="form-container"
    >
      <div>
        <label>
          Type:
        </label>
        <input
          type="text"
          value=""
        />
      </div>
      <div>
        <label>
          Brand:
        </label>
        <input
          type="text"
          value=""
        />
      </div>
      <div>
        <label>
          Owner:
        </label>
        <input
          type="text"
          value=""
        />
      </div>
      <div>
        <label>
          Accessories:
        </label>
        <input
          type="checkbox"
        />
      </div>
      <div>
        <label>
          Warranty:
        </label>
        <input
          type="checkbox"
        />
      </div>
      <div>
        <label>
          Date:
        </label>
        <input
          data-date=""
          data-date-format="DD MMMM YYYY"
          type="date"
        />
      </div>
      <div
        class="buttons"
      >
        <input
          type="button"
          value="Back"
        />
        <input
          type="button"
          value="Save"
        />
      </div>
    </form>
  </div>
</DocumentFragment>
`;

exports[`AddDevice component > renders correctly 1`] = `
<DocumentFragment>
  <div
    class="container"
  >
    <h3
      class="title"
    >
      Add New Device
    </h3>
    <form
      class="form-container"
    >
      <div>
        <label>
          Type:
        </label>
        <input
          type="text"
          value=""
        />
      </div>
      <div>
        <label>
          Brand:
        </label>
        <input
          type="text"
          value=""
        />
      </div>
      <div>
        <label>
          Owner:
        </label>
        <input
          type="text"
          value=""
        />
      </div>
      <div>
        <label>
          Accessories:
        </label>
        <input
          type="checkbox"
        />
      </div>
      <div>
        <label>
          Warranty:
        </label>
        <input
          type="checkbox"
        />
      </div>
      <div>
        <label>
          Date:
        </label>
        <input
          data-date=""
          data-date-format="DD MMMM YYYY"
          type="date"
        />
      </div>
      <div
        class="buttons"
      >
        <input
          type="button"
          value="Back"
        />
        <input
          type="button"
          value="Save"
        />
      </div>
    </form>
  </div>
</DocumentFragment>
`;
